scipy/stats/_continuous_distns.py

```diff
@@ -4346,8 +4346,19 @@ def _stats(self, a, moments='mvsk'):
         return m1, m2, g1, g2
 
     def _entropy(self, a):
-        return a - (a+1.0) * sc.psi(a) + sc.gammaln(a)
+        a = np.asarray([a])
+        def regular(a):
+            h = a - (a + 1.0) * sc.psi(a) + sc.gammaln(a)
+            return h
+
+        def asymptotic(a):
+            # gammaln(a) ~ a * ln(a) - a - 0.5 * ln(a) + 0.5 * ln(2 * pi)
+            # psi(a) ~ ln(a) - 1 / (2 * a)
+            h = -1.5 * np.log(a) + 0.5 * np.log(2 * np.pi) + 0.5 + 1 / (2 * a)
```

Comment:
The powers of `a` can overflow, but that means that the result is negligible. This could be fixed by evaluating them in logspace first, or:
```suggestion
            with np.errstate(over='ignore'):
                h = ((1 - 3*np.log(a) + np.log(2) + np.log(np.pi))/2
                     + 2/(3*a) + 1/(12*a**2) - 1/(90*a**3) - 1/(120*a**4))
            return h
```